app/controllers/amr_heatmap_controller.rb

```diff
@@ -68,169 +69,67 @@ def amr_counts
+    s3_select_params = {
+      bucket: S3_JSON_BUCKET,
+      key: s3_key,
+      expression_type: "SQL",
+      expression: "SELECT * FROM S3Object[*].#{gene_name} LIMIT 1",
+      input_serialization: {
+        json: {
+          type: "DOCUMENT",
+        },
+      },
+      output_serialization: {
+        json: {},
+      },
     }
 
-    # Some drug resistances are only listed in the gene family entry
-    if remote_properties["drugClass"].empty?
-      drug_classes = []
-      remote_properties["geneFamily"].each do |node|
-        gene_family_accession = node.get_attribute("rdf:resource").split('_')[1]
-        drug_classes.concat(get_drug_classes(gene_family_accession, xml_dom))
-      end
-      remote_properties["drugClass"] = drug_classes
-    end
-
-    local_properties = {
-      "synonyms" => matching_entry.xpath("./oboInOwl:hasExactSynonym"),
-      "publications" => matching_entry.at_xpath("./following-sibling::owl:Axiom").xpath("./oboInOwl:hasDbXref"),
-    }
-
-    local_properties.each do |ontology_key, property_nodes|
-      collection = []
-      property_nodes.each do |node|
-        collection.push(node.content)
-      end
-      ontology_information[ontology_key] = collection
-    end
-
-    remote_properties.each do |ontology_key, property_nodes|
-      collection = []
-      property_nodes.each do |node|
-        accession = node.get_attribute("rdf:resource").split('_')[1]
-        accession_class = xml_dom.at_xpath(".//owl:Class[@rdf:about='http://purl.obolibrary.org/obo/ARO_#{accession}']")
-        label = accession_class.at_xpath("./rdfs:label").content
-        description = accession_class.at_xpath("./obo:IAO_0000115").content
-        entry = { "label" => label, "description" => description }
-        collection.push(entry)
+    entry = []
+    begin
+      @client.select_object_content(s3_select_params) do |stream|
+        stream.on_records_event do |event|
+          entry.push(event.payload.read)
+        end
       end
-      ontology_information[ontology_key] = collection
-    end
-    return ontology_information
-  end
-
-  def get_drug_classes(accession, xml_dom)
-    accession_class = xml_dom.at_xpath(".//owl:Class[@rdf:about='http://purl.obolibrary.org/obo/ARO_#{accession}']")
-    drug_class_node_markers = accession_class.xpath("./rdfs:subClassOf/owl:Restriction/owl:onProperty[@rdf:resource='http://purl.obolibrary.org/obo/RO#_confers_resistance_to_drug']")
-    drug_class_family_node_markers = accession_class.xpath("./rdfs:subClassOf/owl:Restriction/owl:onProperty[@rdf:resource='http://purl.obolibrary.org/obo/RO#_confers_resistance_to']")
-    drug_markers = drug_class_node_markers + drug_class_family_node_markers
-    drug_class_nodes = []
-    drug_markers.each do |node_marker|
-      drug_class_nodes.push(node_marker.at_xpath("./following-sibling::owl:someValuesFrom"))
+    rescue
+      Rails.logger.info("Error retrieving ontology entry for #{gene_name} from s3")
```

Comment:
you probably want to LogUtil.log_err_and_airbrake or at least Rails.logger.error these two errors, right?

Reply:
Yeah, will fix this.